src/main/java/seedu/hustler/task/Reminders.java

```diff
@@ -0,0 +1,235 @@
+            boolean checkRange = list.get(i).toString().contains("[R]");
+            if (!done && checkDeadline || checkEvent || checkRange) {
+                if (checkOverdue(i)) {
+                    overDueList.add(list.get(i));
+                }
+            }
+        }
+        return overDueList;
+    }
+
+    /**
+     * List of tasks that are due in 24 hours or less.
+     * @param list current TaskList.
+     * @return a list of tasks that are due in 24 hours or less.
+     */
+    public static ArrayList<Task> lastDay(TaskList list) {
+        lastDayList.clear();
+        for (int i = 0; i < list.size(); i++) {
+            LocalDateTime currentTime = LocalDateTime.now();
+            boolean done = list.get(i).isDone;
+            boolean checkDeadline = list.get(i).toString().contains("[D]");
+            boolean checkEvent = list.get(i).toString().contains("[E]");
+            boolean checkRange = list.get(i).toString().contains("[R]");
+            if (!done && checkDeadline || checkEvent || checkRange) {
+                long duration = Duration.between(currentTime, list.get(i).getDateTime()).getSeconds();
+                if (duration <= TWENTY_FOUR_HOURS && duration > THIRTY_MINUTES) {
+                    lastDayList.add(list.get(i));
+                }
+            }
+        }
+        return lastDayList;
+    }
+
+    /**
+     * List of tasks that are due in 30 minutes or less.
+     * @param list current TaskList.
+     * @return a list of tasks that are due in 30 minutes or less.
+     */
+    public static ArrayList<Task> lastThirtyMins(TaskList list) {
+        lastThirtyMinutesList.clear();
+        for (int i = 0; i < list.size(); i++) {
+            LocalDateTime currentTime = LocalDateTime.now();
+            boolean done = list.get(i).isDone;
+            boolean checkDeadline = list.get(i).toString().contains("[D]");
+            boolean checkEvent = list.get(i).toString().contains("[E]");
+            boolean checkRange = list.get(i).toString().contains("[R]");
+            if (!done && checkDeadline || checkEvent || checkRange) {
+                long duration = Duration.between(currentTime, list.get(i).getDateTime()).getSeconds();
+                if (duration <= THIRTY_MINUTES && duration > 0) {
+                    lastThirtyMinutesList.add(list.get(i));
+                }
+            }
+        }
+        return lastThirtyMinutesList;
+    }
+
+    /**
+     * Runs all types of reminders list at once.
+     * @param list current TaskList.
+     */
+    public static void runAll(TaskList list) {
+        overdue(list);
+        lastThirtyMins(list);
+        lastDay(list);
+    }
+
+    /**
+     * Display reminders list.
+     */
+    public static void displayReminders() {
+        if (!overDueList.isEmpty()) {
+            if (overDueList.size() == 1) {
+                System.out.println("\t_____________________________________");
```

Comment:
These lines can be saved into some variable since they are used so many times. Also to avoid "magic strings"; if you need to change these in the future, it would be a hassle to do so to all of them.